tools/api-linter/src/config.rs

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use serde::de::{SeqAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+use std::fmt;
+use wildmatch::WildMatch;
+
+#[derive(Debug, Deserialize)]
+pub struct Config {
+    /// Whether or not to allow types from `alloc`. Defaults to true.
+    #[serde(default = "default_allow_std")]
+    pub allow_alloc: bool,
+
+    /// Whether or not to allow types from `core`. Defaults to true.
+    #[serde(default = "default_allow_std")]
+    pub allow_core: bool,
+
+    /// Whether or not to allow types from `std`. Defaults to true.
+    #[serde(default = "default_allow_std")]
+    pub allow_std: bool,
+
+    /// List of globs for allowed external types
+    ///
+    /// For example, to allow every type in a crate:
+    /// ```toml
+    /// allowed_external_types = [
+    ///     "crate_name::*"
+    /// ]
+    /// ```
+    ///
+    /// Or, to selectively allow just a module of that crate
+    /// ```toml
+    /// allowed_external_types = [
+    ///     "crate_name::path::to_module::*"
+    /// ]
+    /// ```
+    #[serde(deserialize_with = "deserialize_vec_wild_match")]
+    pub allowed_external_types: Vec<WildMatch>,
```

Comment:
I originally had it as `Vec<String>`, but then I wanted to cache the result of the [pattern parsing that `wildmatch` does](https://github.com/becheran/wildmatch/blob/master/src/lib.rs#L60) rather than run it for every type that needs to be checked. I could probably accomplish the caching with some kind of interior mutability rather than using serde though if you prefer.